smack-extensions/src/main/java/org/jivesoftware/smackx/chatstates/ChatStateManager.java

```diff
@@ -45,17 +54,39 @@
  * If this does not occur you will not receive the update notifications.
  *
  * @author Alexander Wenckus
+ * @author Paul Schaub
  * @see org.jivesoftware.smackx.chatstates.ChatState
  * @see org.jivesoftware.smackx.chatstates.packet.ChatStateExtension
  */
-// TODO Migrate to new chat2 API on Smack 4.3.
-@SuppressWarnings("deprecation")
 public final class ChatStateManager extends Manager {
+
     public static final String NAMESPACE = "http://jabber.org/protocol/chatstates";
 
     private static final Map<XMPPConnection, ChatStateManager> INSTANCES = new WeakHashMap<>();
 
     private static final StanzaFilter filter = new NotFilter(new StanzaExtensionFilter(NAMESPACE));
+    private static final StanzaFilter INCOMING_MESSAGE_FILTER =
+            new AndFilter(MessageTypeFilter.NORMAL_OR_CHAT, FromTypeFilter.ENTITY_FULL_JID);
+
+    /**
+     * Message listener, that appends a ChatStateExtension to outgoing messages
+     */
+    private final OutgoingMessageInterceptor outgoingInterceptor = new OutgoingMessageInterceptor();
```

Comment:
Could you move those down again, so that it is clear that you only added javadoc here